### PR TITLE
add hover effect to PAL

### PR DESCRIPTION
**Pull request by davidkelly**
As you mouseover the various avatars with the PAL up, you will see the sphere and diffuse glow change.  Hoping for feedback about color of sphere, etc...

Working in HMD now too, so this is ready for review, qa...

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: fef8c8900f21f1c847599caf1493439c1768e0fd available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-fef8c89.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-fef8c89.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 409e785aa3d4baeac8d7eb70d5ab5852aeffb77c available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-409e785.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-409e785.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5c6ae6fe95ca1a6d71799f139f8d7484173a8dad available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-5c6ae6f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-5c6ae6f.dmg

File: scripts/system/pal.js [hunk 1]
```diff
@@ -93,14 +120,25 @@ ExtendedOverlay.some = function (iterator) { // Bails early as soon as iterator
...
-ExtendedOverlay.applyPickRay = function (pickRay, cb) { // cb(overlay) on the one overlay intersected by pickRay, if any.
+ExtendedOverlay.unHover = function () { // calls hover(false) on lastHoveringId (if any)
+    if (lastHoveringId) {
+        ExtendedOverlay.get(lastHoveringId).hover(false);
+    }
+};
+
+// hit(overlay) on the one overlay intersected by pickRay, if any.
+// noHit() if no ExtendedOverlay was intersected (helps with hover)
+ExtendedOverlay.applyPickRay = function (pickRay, hit, noHit) { // cb(overlay) on the one overlay intersected by pickRay, if any.
```
Comment (howard-stearns) [inline scripts/system/pal.js]: II guess the comment on 131 should go away.

File: scripts/system/pal.js [hunk 2]
```diff
@@ -311,6 +349,7 @@ function updateOverlays() {
 }
 function removeOverlays() {
     selectedIds = [];
+    lastHoveringId = 0;
```
Comment (howard-stearns) [inline scripts/system/pal.js]: I don't know what the overlay numbers run from. Is 0 ever valid?

File: scripts/system/pal.js (hunk 1 above)
Comment (davidkelly) [inline scripts/system/pal.js]: true!  will do.

File: scripts/system/pal.js (hunk 2 above)
Comment (davidkelly) [inline scripts/system/pal.js]: I will check to be sure.

File: scripts/system/pal.js [hunk 3]
```diff
@@ -338,6 +376,70 @@ function handleMouseEvent(mousePressEvent) { // handleClick if we get one.
...
+                }
+            }
+            currentHandPressed = 0;
+        }
+
+    }
+    return currentHandPressed;
+}
+
+function handleMouseMove(pickRay) { // given the pickRay, just do the hover logic
```
Comment (howard-stearns) [inline scripts/system/pal.js]: My linter doesn't work on windows, but from time to time I do lint javascripts that I'm working on, on mac.
Lint likes functions to be defined before they are used -- or to have the forward reference declared using var or in /*globals...*/ at the top of the file. Could we do that going forward?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 17333ee307288109d2b8101da4d793359fcd0667 available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-17333ee.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-17333ee.dmg

File: scripts/system/pal.js (hunk 3 above)
Comment (davidkelly) [inline scripts/system/pal.js]: ohh - I agree.  I moved things around, and this is the result.  Good catch.

File: scripts/system/pal.js (hunk 2 above)
Comment (davidkelly) [inline scripts/system/pal.js]: Checked `Overlays.h` -- `UNKNOWN_OVERLAY_ID=0` so this is fine.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5fd11ade9cfbe9445480b947d2c4c16fd3627281 available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-5fd11ad.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-5fd11ad.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2d55ed265807eafa43a7d23cd34957ba848f90af available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-2d55ed2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-2d55ed2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4fade3a53eb07db6d2b7d980c832b99194ea606d available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-4fade3a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-4fade3a.dmg

**Comment by nbq** [comment]
Ok, I have tested this and am running the PAL and I notice people that are not selected are this greenish color, when you select them it turns this orange red, to be honest I would see it making more sense if the people (balls) showed up as red first then green once you selected them.

I also mouse over an unselected person and I see the ball turns grey, to me this is odd because when you see grey when you mouse over something in the normal computing world it is not selectable.

I see it in my head as everyone red, when moused over it turns blue, when selected it turns green. If you moused over it once selected, maybe turn an orangish color then.

On a QA level its a PASS as far as mouseover functionality, but since you asked for feedback, I am giving it as a software user just looking at it straight forward.

**Comment by davidkelly** [comment]
Thx @nbq - the feedback is great!  I also didn't really care for the grey sphere when in hover, though I like that the diffuse glow gets brighter.  We will all talk about those things and see what the consensus is - luckily easy to change!  @howard-stearns noted that the hover should occur when we partially press the trigger (blue laser).  I'll make that one change tonight, and await more feedback on look/feel.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0a018e133dfcf63e80a54da41474165dd024d38d available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-0a018e1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-0a018e1.dmg

**Comment by howard-stearns** [comment]
The implementation is different from handControllerPointer, and in the end it works and I can't say one is better than the other.  One specific problem with this, that I do consider a bug, is that you don't get hover-highlighting when using the mouse in HMD.

I'll talk with Seth about what we might do going forward for rationalization...

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1e3a00d1fbb621548e20d856833c411972bb8dc9 available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-1e3a00d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-1e3a00d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e36f6b65212b6f55f61ae93c5b1265507947d172 available at: https://gist.github.com/440971b89498c9213ed1e886f56eef5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-e36f6b6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9337-e36f6b6.dmg

**Pull request merged**